### PR TITLE
Update lock and group button state when selection changed in CanvasItemEditor

### LINKED ISSUE #89016: Lock button sometimes does not update
### Tested versions

4.3 7d2ca2d

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

Apparently Lock button is disabled/enabled in `_draw_viewport()` method, which means that it won't update when selecting nodes does not redraw editor:

https://github.com/godotengine/godot/assets/2223172/7f4aceae-01c0-4bef-bcbb-7851251c6a0c

See how the padlock in the top-right is always disabled until I move cursor over the viewport.

### Steps to reproduce

1. Add Sprite2D
2. Unselect everything
3. Select the Sprite2D
4. Look at Lock button
5. It should be enabled

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/89016

Node3D seems not suffered from this issue.

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -2640,6 +2640,36 @@ void CanvasItemEditor::_update_cursor() {
...
 
+void CanvasItemEditor::_update_lock_and_group_button() {
+	bool all_locked = true;
+	bool all_group = true;
+	List<Node *> selection = editor_selection->get_selected_node_list();
+	if (selection.is_empty()) {
+		all_locked = false;
+		all_group = false;
+	} else {
+		for (Node *E : selection) {
```
Comment (Maran23) [inline editor/plugins/canvas_item_editor_plugin.cpp:2651]: why not use one for loop?

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 2]
```diff
@@ -2640,6 +2640,36 @@ void CanvasItemEditor::_update_cursor() {
...
+				all_locked = false;
+				break;
+			}
+		}
+		for (Node *E : selection) {
+			if (Object::cast_to<CanvasItem>(E) && !Object::cast_to<CanvasItem>(E)->has_meta("_edit_group_")) {
+				all_group = false;
+				break;
+			}
+		}
```
Comment (AThousandShips) [inline editor/plugins/canvas_item_editor_plugin.cpp:2662]:
```suggestion
		for (Node *E : selection) {
			if (Object::cast_to<CanvasItem>(E)) {
				if (all_locked && !Object::cast_to<CanvasItem>(E)->has_meta("_edit_lock_")) {
					all_locked = false;
				}
				if (all_group && !Object::cast_to<CanvasItem>(E)->has_meta("_edit_group_")) {
					all_group = false;
				}
			}
			if (!all_locked && !all_group) {
				break;
			}
		}
```
Something like this

File: editor/plugins/canvas_item_editor_plugin.cpp (hunk 1 above)
Comment (jsjtxietian) [reply editor/plugins/canvas_item_editor_plugin.cpp:2651]: We can use one loop of course, just copied and pasted from the origin code.

**Comment by KoBeWi** [comment]
You could easily supersede the 2D part of #88997 now.

Add a new bool under `all_group` called `has_canvas_item`. Set it to true inside `if (item)` condition and disable buttons based on `has_canvas_item` (empty selection is handled for free).

**Comment by jsjtxietian** [comment]
> Add a new bool under `all_group` called `has_canvas_item`. Set it to true inside `if (item)` condition and disable buttons based on `has_canvas_item` (empty selection is handled for free).

Yes, I guess I can wait for this to be merged first and then rebase that pr to keep commits seperate.

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 3]
```diff
@@ -3796,33 +3828,7 @@ void CanvasItemEditor::_draw_viewport() {
...
-		}
-	}
-
-	lock_button->set_visible(!all_locked);
-	lock_button->set_disabled(selection.is_empty());
-	unlock_button->set_visible(all_locked);
-	group_button->set_visible(!all_group);
-	group_button->set_disabled(selection.is_empty());
-	ungroup_button->set_visible(all_group);
+	_update_lock_and_group_button();
```
[the remark this answers is not in the document]
Comment (jsjtxietian) [reply editor/plugins/canvas_item_editor_plugin.cpp:3831]: Thanks!

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 4]
```diff
@@ -4027,6 +4030,8 @@ void CanvasItemEditor::_notification(int p_what) {
 }
 
 void CanvasItemEditor::_selection_changed() {
+	// hide/show buttons depending on the selection
```
Comment (KoBeWi) [inline editor/plugins/canvas_item_editor_plugin.cpp:4033]:
```suggestion
	// Hide/show buttons depending on the selection.
```

**Review by KoBeWi — APPROVED** [review]

File: editor/plugins/canvas_item_editor_plugin.cpp (hunk 4 above)
Comment (jsjtxietian) [reply editor/plugins/canvas_item_editor_plugin.cpp:4033]: I will delete this.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**